### ISSUE TITLE
Nested polymorphic item support

### ISSUE THREAD

**Issue by autark** [HL7/fhirpath.js#33]
I was trying to apply the new polymorphic item support (using v1.0.0) to a resource of type QuestionnaireResponse. While it seems to be possible to query the polymorhpic value[x] nodes located at the root level using just `value`, nested nodes seems to require one to explicitly write out the value type, eg. `valueString`.

At root level, `answer.value` seems to work as expected:
```
$ node fhirpath -f questionnaireResponse.txt -m stu3 -e "QuestionnaireResponse.item[0].answer.value"
fhirpath(QuestionnaireResponse.item[0].answer.value) =>
[
 "Red"
]
```

But trying to fetch answers for nested items gives no result:
```
$ node fhirpath -f questionnaireResponse.txt -m stu3 -e "QuestionnaireResponse.item[0].item[0].answer.value"
fhirpath(QuestionnaireResponse.item[0].item[0].answer.value) =>
[]

$ node fhirpath -f questionnaireResponse.txt -m stu3 -e "QuestionnaireResponse.item[0].answer.item[0].answer.value"
fhirpath(QuestionnaireResponse.item[0].answer.item[0].answer.value) =>
[]
```

Querying using the specific value type works:
```
$ node fhirpath -f questionnaireResponse.txt -m stu3 -e "QuestionnaireResponse.item[0].item[0].answer.valueString"
fhirpath(QuestionnaireResponse.item[0].item[0].answer.valueString) =>
[
 "Blue"
]
```

Also, using some functions seems to throw it off, f.ex. while this works:
```
$ node fhirpath -f questionnaireResponse.txt -m stu3 -e "QuestionnaireResponse.item.where(linkId='1').answer.value"
fhirpath(QuestionnaireResponse.item.where(linkId='1').answer.value) =>
[
 "Red"
]
```

a similar query like the following doesn't even though the node is at the root level:
```
$ node fhirpath -f questionnaireResponse.txt -m stu3 -e "QuestionnaireResponse.descendants().where(linkId='1').answer.value"
fhirpath(QuestionnaireResponse.descendants().where(linkId='1').answer.value) =>
[]
```
[questionnaireResponse.txt](https://github.com/HL7/fhirpath.js/files/4044723/questionnaireResponse.txt)

**Comment by plynchnlm** [comment]
Thanks for the report.  I did not consider recursive definitions (item.item and item.answer.item) in my implementation.   I will have to find a way to handle that.

I am not sure offhand why the path with "descendants()" is failing, but I will look into that too.

**Comment by niquola** [comment]
The root of the problem - https://github.com/fhir-fuel/fhir-fuel.github.io/issues/2
We should fix it in FHIR JSON representation!

**Comment by plynchnlm** [comment]
Fixed in 1.0.1.
